controllers/utils/storageclass.go

```diff
@@ -125,7 +106,12 @@ func CalculateStorageId(ctx context.Context, c client.Client, sc *storagev1.Stor
 
 	switch sc.Provisioner {
 	case fmt.Sprintf(RBDProvisionerTemplate, storageClusterNamespacedName.Namespace):
-		storageId = CalculateMD5Hash([2]string{cephClusterFSID, DefaultRadosNamespace})
+		radosNamespaceName := DefaultRadosNamespace
+		rbdRadosNamespace := fmt.Sprintf(DefaultRadosNamespaceStorageClassTemplate, storageClusterNamespacedName.Name)
+		if strings.HasPrefix(sc.Name, rbdRadosNamespace) {
+			radosNamespaceName = strings.TrimPrefix(sc.Name, rbdRadosNamespace)
```

Comment:
Is this correct ? It will just  return the SC name without the `%s-ceph-rbd-rados-namespace-{rest of sc.Name}` . That is supposed to contain the radosNamespacedName ?

https://cs.opensource.google/go/go/+/go1.24.2:src/strings/strings.go;l=1125

Reply:
That is expected. I am retrieving RadosNamesapce name from StorageClass name.